controller/array_action/array_mediator_svc.py

```diff
@@ -1292,3 +1309,58 @@ def _get_host_name_if_equal(self, nvme_host, fc_host, iscsi_host):
         if len(unique_names) == 1:
             return unique_names.pop()
         return None
+
+    def _addsnapshot(self, name, source_volume_id, pool):
+        try:
+            return self.client.svctask.addsnapshot(name=name, volumes=source_volume_id, pool=pool)
+        except (svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            if is_warning_message(ex.my_message):
+                logger.warning("exception encountered while creating snapshot '{}': {}".format(name,
+                                                                                               ex.my_message))
+            else:
+                logger.error("cannot create snapshot {0}, Reason is: {1}".format(name, ex))
+                if OBJ_ALREADY_EXIST in ex.my_message:
+                    raise array_errors.SnapshotAlreadyExists(name, self.endpoint)
```

Comment:
does this mean we [don't](https://github.com/IBM/ibm-block-csi-driver/blob/v1.9.0/controller/controller_server/csi_controller_server.py#L382) [return](https://github.com/IBM/ibm-block-csi-driver/blob/v1.9.0/controller/array_action/array_mediator_svc.py#L468) the correct status code for flashcopy 1.0 snapshots that already exist?

Reply:
done